src/vmm/src/vstate.rs

```diff
@@ -960,8 +960,12 @@ impl Vcpu {
             .fd
             .get_vcpu_events()
             .map_err(Error::VcpuGetVcpuEvents)?;
+
         Ok(VcpuState {
-            cpuid: self.cpuid.clone(),
+            cpuid: self
+                .fd
+                .get_cpuid2(kvm_bindings::KVM_MAX_CPUID_ENTRIES)
```

Comment:
Are there any ordering requirements between `GET_CPUID` and the other `GET_*` ? If there are and we don't follow them we could introduce subtle and sporadic failures.

Reply:
AFAIK there are no explicit requirements here.